In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np
df=pd.read_csv('data/stud.csv')

In [2]:
df.head(2)

,gender,race_ethnicity,parental_level_of_education,lunch,test_preparation_course,math_score,reading_score,writing_score
0,female,group B,bachelor's degree,standard,none,72,72,74
1,female,group C,some college,standard,completed,69,90,88


In [24]:
df.shape

(1000, 8)

In [5]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 8 columns):
 #   Column                       Non-Null Count  Dtype
---  ------                       --------------  -----
 0   gender                       1000 non-null   str  
 1   race_ethnicity               1000 non-null   str  
 2   parental_level_of_education  1000 non-null   str  
 3   lunch                        1000 non-null   str  
 4   test_preparation_course      1000 non-null   str  
 5   math_score                   1000 non-null   int64
 6   reading_score                1000 non-null   int64
 7   writing_score                1000 non-null   int64
dtypes: int64(3), str(5)
memory usage: 103.5 KB


In [ ]:
df['gender'].unique()

<ArrowStringArray>
['female', 'male']
Length: 2, dtype: str

In [ ]:
df['race_ethnicity'].unique()

<ArrowStringArray>
['group B', 'group C', 'group A', 'group D', 'group E']
Length: 5, dtype: str

In [34]:
df['parental_level_of_education'].unique()

<ArrowStringArray>
[ 'bachelor's degree',       'some college',    'master's degree',
 'associate's degree',        'high school',   'some high school']
Length: 6, dtype: str

In [35]:
df['lunch'].unique()

<ArrowStringArray>
['standard', 'free/reduced']
Length: 2, dtype: str

In [36]:
df['test_preparation_course'].unique()

<ArrowStringArray>
['none', 'completed']
Length: 2, dtype: str

In [38]:
df.describe()

,math_score,reading_score,writing_score
count,1000.00000,1000.000000,1000.000000
mean,66.08900,69.169000,68.054000
std,15.16308,14.600192,15.195657
min,0.00000,17.000000,10.000000
25%,57.00000,59.000000,57.750000
50%,66.00000,70.000000,69.000000
75%,77.00000,79.000000,79.000000
max,100.00000,100.000000,100.000000


In [39]:
df.isnull().sum()

gender                         0
race_ethnicity                 0
parental_level_of_education    0
lunch                          0
test_preparation_course        0
math_score                     0
reading_score                  0
writing_score                  0
dtype: int64

# There are no missing values in the dataset

In [40]:
df.duplicated().sum()

np.int64(0)

# There are no duplicated rows

Checking the datatypes

In [41]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 8 columns):
 #   Column                       Non-Null Count  Dtype
---  ------                       --------------  -----
 0   gender                       1000 non-null   str  
 1   race_ethnicity               1000 non-null   str  
 2   parental_level_of_education  1000 non-null   str  
 3   lunch                        1000 non-null   str  
 4   test_preparation_course      1000 non-null   str  
 5   math_score                   1000 non-null   int64
 6   reading_score                1000 non-null   int64
 7   writing_score                1000 non-null   int64
dtypes: int64(3), str(5)
memory usage: 103.5 KB


Checking the number of unique values in each column

In [43]:
df.nunique()

gender                          2
race_ethnicity                  5
parental_level_of_education     6
lunch                           2
test_preparation_course         2
math_score                     81
reading_score                  72
writing_score                  77
dtype: int64

Checking the statics of dataset

In [44]:
df.describe()

,math_score,reading_score,writing_score
count,1000.00000,1000.000000,1000.000000
mean,66.08900,69.169000,68.054000
std,15.16308,14.600192,15.195657
min,0.00000,17.000000,10.000000
25%,57.00000,59.000000,57.750000
50%,66.00000,70.000000,69.000000
75%,77.00000,79.000000,79.000000
max,100.00000,100.000000,100.000000


### Insight
##### . From the above distribution of numerical data, all means are very close to each other - between 66 and 69.17
##### . All Standard deviations are also close - between 14.6 and 15.19
##### . Although minimum value for math_score is 0, for reading_score =10 , for writing_score minimum is 17

# EDA

In [ ]:
categorical=[i for i in df.columns if df[i].dtype=='str']
numerical=[i for i in df.columns if i not in categorical]
discrete=[i for i in numerical if len(df[i])<=25]
continuous=[i for i in numerical if i not in discrete]

['math_score', 'reading_score', 'writing_score']

In [60]:
for i in categorical:
    print(f'For {i}: ',end=' ')
    print(df[i].unique().tolist())

For gender:  ['female', 'male']
For race_ethnicity:  ['group B', 'group C', 'group A', 'group D', 'group E']
For parental_level_of_education:  ["bachelor's degree", 'some college', "master's degree", "associate's degree", 'high school', 'some high school']
For lunch:  ['standard', 'free/reduced']
For test_preparation_course:  ['none', 'completed']
